demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,625 @@
+
+/**
+ * @brief Unix time.
+ *
+ * @return Time in milliseconds.
+ */
+uint64_t ullGetUnixTime( void );
+/*-----------------------------------------------------------*/
+
+/* Define buffer for IoTHub info.  */
+#ifdef democonfigENABLE_DPS_SAMPLE
+    static uint8_t ucSampleIotHubHostname[ 128 ];
+    static uint8_t ucSampleIotHubDeviceId[ 128 ];
+    static AzureIoTProvisioningClient_t xAzureIoTProvisioningClient;
+#endif /* democonfigENABLE_DPS_SAMPLE */
+
+static uint8_t ucPropertyBuffer[ 32 ];
+
+/* Each compilation unit must define the NetworkContext struct. */
+struct NetworkContext
+{
+    TlsTransportParams_t * pParams;
+};
+
+static AzureIoTHubClient_t xAzureIoTHubClient;
+/*-----------------------------------------------------------*/
+
+#ifdef democonfigENABLE_DPS_SAMPLE
+
+/**
+ * @brief Gets the IoTHub endpoint and deviceId from Provisioning service.
+ *
+ * @param[in] pXNetworkCredentials  Network credential used to connect to Provisioning service
+ * @param[out] ppucIothubHostname  Pointer to uint8_t* IoTHub hostname return from Provisioning Service
+ * @param[in,out] pulIothubHostnameLength  Length of hostname
+ * @param[out] ppucIothubDeviceId  Pointer to uint8_t* deviceId return from Provisioning Service
+ * @param[in,out] pulIothubDeviceIdLength  Length of deviceId
+ */
+    static uint32_t prvIoTHubInfoGet( NetworkCredentials_t * pXNetworkCredentials,
+                                      uint8_t ** ppucIothubHostname,
+                                      uint32_t * pulIothubHostnameLength,
+                                      uint8_t ** ppucIothubDeviceId,
+                                      uint32_t * pulIothubDeviceIdLength );
+
+#endif /* democonfigENABLE_DPS_SAMPLE */
+
+/**
+ * @brief The task used to demonstrate the MQTT API.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+static void prvAzureDemoTask( void * pvParameters );
+
+
+/**
+ * @brief Connect to MQTT broker with reconnection retries.
+ *
+ * If connection fails, retry is attempted after a timeout.
+ * Timeout value will exponentially increase until maximum
+ * timeout value is reached or the number of attempts are exhausted.
+ *
+ * @param[out] pxNetworkContext The parameter to return the created network context.
+ *
+ * @return The status of the final connection attempt.
+ */
+static TlsTransportStatus_t prvConnectToServerWithBackoffRetries( const char * pcHostName,
+                                                                  uint32_t ulPort,
+                                                                  NetworkCredentials_t * pxNetworkCredentials,
+                                                                  NetworkContext_t * pxNetworkContext );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static uint8_t ucSharedBuffer[ democonfigNETWORK_BUFFER_SIZE ];
```

Comment:
If comment is correct and it's used for MQTT messages *only* could we rename to like `ucMqttMessageBuffer` or something more descriptive?  Or if it's used all over the place we should update comment & leave naming as-is.